tests/app/questionnaire/test_path_finder.py

```diff
@@ -535,3 +536,221 @@ def test_new_remove_answer_and_block_if_routing_backwards(list_store):
+@pytest.mark.parametrize(
+    "schema",
+    (
+        [
+            "test_new_routing_and_skipping_section_dependencies",
+            "test_routing_and_skipping_section_dependencies",
+        ]
+    ),
+)
+def test_routing_path_block_ids_dependent_on_other_sections_when_rules(
+    list_store,
+    skip_age_answer,
+    skip_confirmation_answer,
+    schema,
+    section_id,
+    expected_route,
+):
+    # Given a schema which has when rules in a section which has dependencies on other sections answers
+    schema = load_schema_from_name(schema)
+    answer_store = AnswerStore()
+    answer_store.add_or_update(
+        Answer(answer_id="skip-age-answer", value=skip_age_answer)
+    )
+
+    progress = [
+        {
+            "section_id": "skip-section",
+            "list_item_id": None,
+            "status": CompletionStatus.COMPLETED,
+            "block_ids": ["skip-age"],
+        }
+    ]
+
+    if skip_confirmation_answer:
+        answer_store.add_or_update(
+            Answer(answer_id="skip-confirmation-answer", value=skip_confirmation_answer)
+        )
+        answer_store.add_or_update(Answer(answer_id="security-answer", value="Yes"))
+
+        progress.append(
+            {
+                "section_id": "skip-confirmation-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["security", "skip-confirmation"],
+            }
+        )
+
+    progress_store = ProgressStore(progress)
+
+    # When I build the path
+    path_finder = PathFinder(
+        schema,
+        answer_store,
+        list_store,
+        progress_store,
+        metadata={},
+        response_metadata={},
+    )
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    # Then the path is built correctly
+    expected_routing_path = RoutingPath(
+        expected_route,
+        section_id=section_id,
+    )
+    assert routing_path == expected_routing_path
+
+
+@pytest.mark.parametrize(
+    "skip_age_answer, expected_route",
+    (
+        (
+            # Answering 'Yes' to the skip age question
+            # means in all repeating sections you won't be asked their age
+            "Yes",
+            ["repeating-sex"],
+        ),
+        (
+            # Answering 'No' to the skip age question
+            # means in all repeating sections you will be asked their age
+            "No",
+            ["repeating-sex", "repeating-age"],
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "schema",
+    (
+        [
+            "test_new_routing_and_skipping_section_dependencies",
+            "test_routing_and_skipping_section_dependencies",
+        ]
+    ),
+)
+def test_routing_path_block_ids_dependent_on_other_sections_when_rules_repeating(
+    skip_age_answer,
+    schema,
+    expected_route,
+):
+    # Given a schema with repeating sections which has when rules dependent on another section
+    schema = load_schema_from_name(schema)
```

Comment:
This clobbers the arg to the test function

Reply:
Sorry, I don't have any idea what this comment means